dkron/job.go

```diff
@@ -396,3 +399,68 @@ func isSlug(candidate string) (bool, string) {
 	whyNot := illegalCharPattern.FindString(candidate)
 	return whyNot == "", whyNot
 }
+
+// generate Job Tree
+func generateJobTree(jobs []*Job) ([]*Job, error) {
+	length := len(jobs)
+	j := 0
+	for i := 0; i < length; i++ {
+		rejobs, isTopParentNodeFlag, err := findParentJobAndValidateJob(jobs, j)
+		if err != nil {
+			return nil, err
+		}
+		if isTopParentNodeFlag {
+			j++
+		}
+		jobs = rejobs
+	}
+	return jobs, nil
+}
+
+// find parent job and validate job
```

Comment:
This is not correctly linted, it should start with `findParentJobAndValidateJob...`